dcbr-api/api/models.py

```diff
@@ -97,6 +105,69 @@ class Address(models.Model):
     PRIMARY = "PRI"
     OPERATION = "OPN"
     TYPE_CHOICES = ((PRIMARY, "Primary"), (OPERATION, "Operation"))
+
+    # List of regions
+    ALBERNI_CLAYOQUOT = "Alberni-Clayoquot"
+    BULKLEY_NECHAKO = "Bulkley-Nechako"
+    CAPITAL = "Capital"
+    CARIBOO = "Cariboo"
+    CENTRAL_COAST = "Central Coast"
+    CENTRAL_KOOTENAY = "Central Kootenay"
+    CENTRAL_OKANAGAN = "Central Okanagan"
+    COLUMBIA_SHUSWAP = "Columnbia Shuswap"
+    COMOX_VALLEY = "Comox Valley"
+    COWICHAN_VALLEY = "Cowichan Valley"
+    EAST_KOOTENAY = "East Kootenay"
+    FRASER_VALLEY = "Fraser Valley"
+    FRASER_FORT_GEORGE = "Fraser-Fort George"
+    ISLANDS_TRUST = "Islands Trust"
+    KITIMAT_STIKINE = "Kitimat Stikine"
+    KOOTENAY_BOUNDARY = "Kootenay Boundary"
+    METRO_VANCOUVER = "Metro Vancouver"
+    MOUNT_WADDINGTON = "Mount Waddington"
+    NANAIMO = "Nanaimo"
+    NORTH_OKANAGAN = "North Okanagan"
+    NORTH_COAST = "North Coast"
+    OKANAGAN_SIMILKAMEEN = "Okanagan-Similkameen"
+    PEACE_RIVER = "Peace River"
+    GATHET = "Gathet"
+    SQUAMISH_LILLOOET = "Squamish-Lillooet"
+    STRATHCONA = "Strathcona"
+    SUNSHINE_COAST = "Sunshine Coast"
+    THOMPSON_NICOLA = "Thompson-Nicola"
+
+    TYPE_CHOICES = ((PRIMARY, "Primary"), (OPERATION, "Operation"))
+    REGIONAL_CHOICES = (
+        (ALBERNI_CLAYOQUOT, "Alberni-Clayoquot"),
+        (BULKLEY_NECHAKO, "Buckley-Nechako"),
+        (CAPITAL, "Capital"),
+        (CARIBOO, "Cariboo"),
+        (CENTRAL_COAST, "Central Coast"),
+        (CENTRAL_KOOTENAY, "Central Kootenay"),
+        (CENTRAL_OKANAGAN, "Central Okanagan"),
+        (COLUMBIA_SHUSWAP, "Columbia Shuswap"),
+        (COMOX_VALLEY, "Comox Valley"),
+        (COWICHAN_VALLEY, "Cowichan Valley"),
+        (EAST_KOOTENAY, "East Kootenay"),
+        (FRASER_VALLEY, "Fraser Valley"),
+        (FRASER_FORT_GEORGE, "Fraser-Fort George"),
+        (ISLANDS_TRUST, "Islands Trust"),
+        (KITIMAT_STIKINE,"Kitimat-Stikine"),
+        (KOOTENAY_BOUNDARY, "Kootenay-Boundary"),
+        (METRO_VANCOUVER, "Metro Vancouver"),
+        (MOUNT_WADDINGTON, "Mount Waddington"),
+        (NANAIMO, "Nanaimo"),
+        (NORTH_OKANAGAN, "North Okanagan"),
+        (NORTH_COAST, "North Coast"),
+        (OKANAGAN_SIMILKAMEEN, "Okanagan-Similkameen"),
+        (PEACE_RIVER, "Peace River"),
+        (GATHET, "qathet"),
+        (SQUAMISH_LILLOOET, "Squamish-Lillooet"),
+        (STRATHCONA, "Strathcona"),
+        (SUNSHINE_COAST, "Sunshine Coast"),
+        (THOMPSON_NICOLA, "Thompson-Nicola")
+    )
+
```

Comment:
Why are we declaring the choices twice? Could we use an enum instead? See: https://docs.python.org/3/library/enum.html

It would also be good practice to create a Model that holds the values, and pull it at runtime. We can pre-populate the list via a custom migration (see the `email_service` for an example) and this way, if the region list was going to change for any reason, the admin could just go in and deal with it. To do this, we will also need an API endpoint to GET the data from the UI (we don't want to have out-of-sync entries between the back-end and the front-end).

It should not be a big change, if deemed not in scope for this sprint/ticket please log a new Jira issue to deal with it and include other lists we currently have in the models/UI. They should all be managed as enums in the backend, and we could either have separate models, or a single model called "mappings" with a column used to decode the types (region, contact_method, animal_type, etc.).

Reply:
By twice, do you mean in the models and in the front-end?

Reply:
Yes, the same list of values is declared in both places and can get out of sync.